src/link/Coff.zig

```diff
@@ -1095,6 +1099,10 @@ fn linkWithLLD(self: *Coff, comp: *Compilation) !void {
             try argv.append(p);
         }
 
+        if (self.base.options.out_implib != null) {
+            try argv.append(try allocPrint(arena, "-IMPLIB:{s}.lib", .{full_out_path}));
```

Comment:
Should this not set `-IMPLIB` to the value in `out_implib` rather than `full_out_path`? What if the user sets `--out-implib` to something different than location of the final binary?

Reply:
`full_out_path` here is the path within the cache and not the final destination. The final destination is handled is the path stored in `out_implib` that is used during the `UpdateModule` step, similar to pdbs. We just choose a path here that is unlikely to collide with other linker outputs.

Reply:
Gotcha, thanks.

Reply:
I guess this is fine to leave as-is for now but this will probably need to be cleaned up and put as logic in the linker proper rather than `Compilation.updateModule`. After all, we also need to update a dSYM bundle in MachO which should not be of any concern to the `Compilation` module. cc @andrewrk